src/Agda2Hs/Compile/Function.hs

```diff
@@ -162,10 +161,24 @@ compilePats :: NAPs -> C [Hs.Pat ()]
 compilePats ps = mapM (compilePat . namedArg) =<< filterM keepPat ps
   where
     keepPat :: NamedArg DeBruijnPattern -> C Bool
-    keepPat p = andM
-      [ return $ keepArg p
-      , not <$> isUnboxCopattern (namedArg p)
-      ]
+    keepPat p = do
+      keep <- return (keepArg p) `and2M` (not <$> isUnboxCopattern (namedArg p))
+      when keep $ noAsPatterns $ namedArg p
+      -- We do not allow forced (dot) patterns for non-erased arguments (see issue #142).
+      when (usableModality p && isForcedPat (namedArg p)) $
+        genericDocError =<< "not supported by Agda2Hs: forced (dot) patterns in non-erased positions"
```

Comment:
Can you use the same capitalization of agda2hs (as `Compile/Type.hs:138`) for both error messages?

Reply:
Done!